dot/sync/chain_processor.go

```diff
@@ -42,24 +44,35 @@ type chainProcessor struct {
 	telemetry          telemetry.Client
 }
 
-func newChainProcessor(readyBlocks *blockQueue, pendingBlocks DisjointBlockSet,
```

Comment:
`chainProcessorConfig` and `chainProcessor` are basically the same struct apart from ctx and cancel. `chainProcessorConfig` isn't useful apart from providing better indentation.
I would suggest to remove `chainProcessorConfig` and modify `newChainProcessor` everywhere to provide better indentation.

Like given below, (here and everywhere `newChainProcessor` is used)
```
func newChainProcessor(
	readyBlocks *blockQueue, 
	pendingBlocks DisjointBlockSet,
	blockState BlockState, 
	storageState StorageState,
	transactionState TransactionState, 
	babeVerifier BabeVerifier,
	finalityGadget FinalityGadget, 
	blockImportHandler BlockImportHandler, 
	telemetry telemetry.Client) 
*chainProcessor {
```

Reply:
:warning: repeating myself, but these are out of scope changes, we should have them in a separate ez PR.

Eclesio's [argument](https://github.com/ChainSafe/gossamer/pull/2906#discussion_r1010818200) was:

> I will keep it as it is because we already use a configuration struct for the chain_sync so adding this configuration struct for chain_processor keep them in the same pattern. I have changed to accept a value instead of a pointer.

Although I agree we should standardize our codebase as we progress through it, I agree with Kishan's suggestion especially for unexported/internal constructors.

I would also add in my very opinionated opinion that config structs should be reserved for the public Go API (so exported public constructors) and every field can be set to a default (so the user can pass an empty struct). It also allows to add fields without breaking the constructor signature (so it stays retro-compat). 

For internal constructors, it's far easier to have plain and simple compulsory named arguments which would detect compilation problems when adding an argument.